sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SQLHadoopMapReduceCommitProtocol.scala

```diff
@@ -66,4 +68,18 @@ class SQLHadoopMapReduceCommitProtocol(
     logInfo(s"Using output committer class ${committer.getClass.getCanonicalName}")
     committer
   }
+
+  override def newTaskTempFile(
+      taskContext: TaskAttemptContext, dir: Option[String], ext: String): String = {
+    val path = super.newTaskTempFile(taskContext, dir, ext)
+    if (dynamicPartitionOverwrite) {
+      val numParts = partitionPaths.size
+      if (numParts > maxDynamicPartitions) {
```

Comment:
partitionPaths contains only the paths on a specific Spark task, isn't?

Reply:
Oh, you are right. `partitionPaths` may be similar to `hive.exec.max.dynamic.partitions.pernode`. And the implementation of total limitation has to add a `var totalPartitions: Int` for checking.

Reply:
@viirya fixed